test/test_integration_range_scan.cxx

```diff
@@ -0,0 +1,699 @@
+    };
+
+    auto mutations = populate_documents_for_range_scan(collection, ids, value);
+
+    auto highest_mutation = std::max_element(
+      mutations.begin(), mutations.end(), [](auto a, auto b) { return a.second.sequence_number() < b.second.sequence_number(); });
+
+    auto ag = couchbase::core::agent_group(integration.io, { { integration.cluster } });
+    ag.open_bucket(integration.ctx.bucket);
+    auto agent = ag.get_agent(integration.ctx.bucket);
+    REQUIRE(agent.has_value());
+
+    const std::uint16_t vbucket_id{ 12 };
+    std::vector<std::byte> scan_uuid;
+
+    {
+        couchbase::core::range_scan_create_options options{};
+        options.scope_name = couchbase::scope::default_name;
+        options.collection_name = couchbase::collection::default_name;
+        options.ids_only = true;
+        options.scan_type = couchbase::core::range_scan{
+            { couchbase::core::utils::to_binary("rangescancancel") },
+            { couchbase::core::utils::to_binary("rangescancancel\xff") },
+        };
+        options.snapshot_requirements = couchbase::core::range_snapshot_requirements{
+            highest_mutation->second.partition_uuid(),
+            highest_mutation->second.sequence_number(),
+        };
+
+        auto barrier = std::make_shared<std::promise<std::pair<couchbase::core::range_scan_create_result, std::error_code>>>();
+        auto f = barrier->get_future();
+
+        auto op = agent->range_scan_create(vbucket_id, options, [barrier](auto res, auto ec) {
+            barrier->set_value({ std::move(res), ec });
+        });
+        EXPECT_SUCCESS(op);
+
+        auto [res, ec] = f.get();
+        REQUIRE_SUCCESS(ec);
+        REQUIRE_FALSE(res.scan_uuid.empty());
+        scan_uuid = res.scan_uuid;
+    }
+
+    {
+        auto barrier = std::make_shared<std::promise<std::pair<couchbase::core::range_scan_cancel_result, std::error_code>>>();
+        auto f = barrier->get_future();
+
+        auto op = agent->range_scan_cancel(scan_uuid, vbucket_id, {}, [barrier](auto res, auto ec) {
+            barrier->set_value({ std::move(res), ec });
+        });
+        EXPECT_SUCCESS(op);
+
+        auto [resp, ec] = f.get();
+        REQUIRE_SUCCESS(ec);
+    }
+
+    bool items_callback_invoked{ false };
+    {
+        auto barrier = std::make_shared<std::promise<std::pair<couchbase::core::range_scan_continue_result, std::error_code>>>();
+        auto f = barrier->get_future();
+
+        auto op = agent->range_scan_continue(
+          scan_uuid,
+          vbucket_id,
+          {},
+          [&items_callback_invoked](auto) { items_callback_invoked = true; },
+          [barrier](auto res, auto ec) {
+              barrier->set_value({ std::move(res), ec });
+          });
+        EXPECT_SUCCESS(op);
+
+        auto [resp, ec] = f.get();
+        REQUIRE(ec == couchbase::errc::key_value::document_not_found);
```

Comment:
Should this be a range scan cancelled error code?

Reply:
This is what the server gives us. I guess it does not differentiate "scan_uuid does not exist" from "scan_uuid has been canceled". Or does it, but the test cannot catch it.